.github/workflows/publish.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish Action
+
+on:
+  push:
+    branches:
+      - main
+      - next
+      - alpha
+      - beta
+      # N.x (maintenance release branches)
+      - '[0-9]+.x'
+
+env:
+  NODE_VERSION: 14.x
+
+jobs:
+  build:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2.3.4
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v2.4.0
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: yarn
+
+      - name: Configure Package Manager
+        run: |
+          echo Configuring NPM_PUBLISH_TOKEN globally for .npmrc
+          npm config set '//registry.npmjs.org/:_authToken' ${{ secrets.NPM_PUBLISH_TOKEN }}
+          npm whoami
+
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Test
+        run: yarn test --coverage --silent
+
+      - uses: codecov/codecov-action@v2.0.2
```

Comment:
We shouldn't use Codecov moving forward because they are too expensive to add more user seats. Instead, let's add a TODO to come back to adding support for publishing our coverage results.